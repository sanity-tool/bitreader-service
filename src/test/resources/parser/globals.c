static int *I_P, I;

static void writeToGlobalInt() {
    I = 1;
}

static void writeToGlobalIntPointer() {
    I_P = 0;
}