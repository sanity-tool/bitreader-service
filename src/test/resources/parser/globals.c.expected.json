{
  "functions" : [ {
    "blocks" : [ {
      "instructions" : [ {
        "kind" : "LLVMStore",
        "operands" : [ {
          "kind" : "LLVMConstantIntValueKind"
        }, {
          "kind" : "LLVMGlobalVariableValueKind"
        } ]
      }, {
        "kind" : "LLVMRet"
      } ]
    } ],
    "entryBlockIndex" : 0
  }, {
    "blocks" : [ {
      "instructions" : [ {
        "kind" : "LLVMStore",
        "operands" : [ {
          "kind" : "LLVMConstantPointerNullValueKind"
        }, {
          "kind" : "LLVMGlobalVariableValueKind"
        } ]
      }, {
        "kind" : "LLVMRet"
      } ]
    } ],
    "entryBlockIndex" : 0
  } ]
}